src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeVTableNode.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+using System;
+using System.Collections.Generic;
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    internal class EETypeVTableNode : EmbeddedObjectNode
+    {
+        TypeDesc _type;
+        private bool ShouldBuildFullVTable { get; set; }
+        List<MethodDesc> _slots = new List<MethodDesc>();
+
+        public EETypeVTableNode(TypeDesc type)
+        {
+            _type = type;
+        }
+
+        protected override void OnMarked(NodeFactory context)
+        {
+            if (context.CompilationModuleGroup.ShouldProduceFullType(_type))
+            {
+                ShouldBuildFullVTable = true;
```

Comment:
I did it this way because once a node is marked, it's basically locked in and used in compilation.  Upgrading the VTable to a full type forces all slots to be live which triggers all the conditional dependencies that various types deriving from that VTable's type place on their virtual methods. Basically I'm trying to limit unnecessary expansion incase this VTable turns out to be unnecessary (never gets marked).